src/layers.cpp

```diff
@@ -115,12 +115,15 @@ Layer_::updateActiveLayers(void) {
 }
 
 void Layer_::updateHighestLayer(void) {
-  for (int8_t i = 31; i >= 0; i--) {
+  // If layer_count is set, start there, otherwise search from the
+  // highest possible layer for the top active layer
+  for (int8_t i = (layer_count - 1); i > 0; i++) {
     if (bitRead(LayerState, i)) {
       highestLayer = i;
       return;
     }
   }
+  // return 0 if no higher active layers were found
```

Comment:
It's not exactly _returning_ here. (And the "what" of the line of code is clear, but the "why" could probably use a bit more explanation.)

Reply:
Right. As for the _why_ – that's a good question. I see that it's not actually possible to turn off the default layer (guaranteed by `updateActiveLayers()`). `updateHighestLayer()` should probably be adjusted to not bother counting down past the default layer (it could be changed, so it's not guaranteed to be 0). Unless you want to allow shutting off the default layer, which seems a bit risky.